src/averageAges.js

```diff
@@ -37,7 +43,22 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren ? people.filter(mother => {
+    if (people.find(child => mother.name === child.mother)) {
+      return true;
+    } else {
+      return false;
+    }
+  })
+    : people.filter(person => person.sex === 'f');
```

Comment:
why do you need this `if ... else...` here?
```suggestion
  const women = withChildren
    ? people.filter(mother => people.find(child => mother.name === child.mother))
    : people.filter(person => person.sex === 'f');
```